pkg/api/handlers/catalogs.go

```diff
@@ -0,0 +1,102 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/obot-platform/obot/pkg/api"
+	"github.com/obot-platform/obot/pkg/controller/handlers/toolreference"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CatalogHandler struct {
+	// toolRefHandler is used to force refresh the catalogs.
+	toolRefHandler *toolreference.Handler
+}
+
+func NewCatalogHandler(toolRefHandler *toolreference.Handler) *CatalogHandler {
+	return &CatalogHandler{
+		toolRefHandler: toolRefHandler,
+	}
+}
+
+// List returns all catalogs.
+func (h *CatalogHandler) List(req api.Context) error {
+	var list v1.CatalogList
+	if err := req.List(&list); err != nil {
+		return fmt.Errorf("failed to list catalogs: %w", err)
+	}
+	return req.Write(list)
+}
+
+// Get returns a specific catalog by ID.
+func (h *CatalogHandler) Get(req api.Context) error {
+	var catalog v1.Catalog
+	if err := req.Get(&catalog, req.PathValue("catalog_id")); err != nil {
+		return fmt.Errorf("failed to get catalog: %w", err)
+	}
+	return req.Write(catalog)
+}
+
+// Create creates a new catalog.
+func (h *CatalogHandler) Create(req api.Context) error {
+	urlBytes, err := io.ReadAll(req.Request.Body)
+	if err != nil {
+		return fmt.Errorf("failed to read catalog data: %w", err)
+	}
+
+	// Validate the URL.
+	u, err := url.Parse(string(urlBytes))
+	if err != nil {
+		return fmt.Errorf("invalid URL: %w", err)
+	}
+
+	if u.Scheme != "https" {
+		return fmt.Errorf("only HTTPS URLs are supported")
+	}
+
+	catalog := &v1.Catalog{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: system.CatalogPrefix,
+		},
+		Spec: v1.CatalogSpec{
+			URL: u.String(),
+		},
+	}
+
+	if err := req.Create(catalog); err != nil {
+		return fmt.Errorf("failed to create catalog: %w", err)
+	}
+
+	go func() {
+		if err := h.toolRefHandler.ForceRefreshMCPCatalogs(context.Background(), req.Storage); err != nil {
+			log.Errorf("Failed to force refresh MCP catalogs: %v", err)
+		}
+	}()
```

Comment:
This was the best I could come up with to immediately refresh the catalogs when we create a new one. The way that we read the catalogs right now, we basically ditch everything we have locally, load all the catalogs, and then save that. I felt it would be overkill to refactor this to load individual catalogs at this time (as this would, among other things, involve tying specific `MCPServerCatalogEntries` to their particular `Catalog`), but we could revisit that in the future if we think it is worthwhile.

Reply:
Refactored into a kick pattern with a channel